nautobot/ipam/models.py

```diff
@@ -648,19 +648,18 @@ def get_parent(self):
         return None
 
     def clean(self):
-        if self.prefix:
-            # self.parent depends on self.prefix having a value
-            self.parent = self.get_parent()
-        super().clean()
-
-    def save(self, *args, **kwargs):
-        if isinstance(self.prefix, netaddr.IPNetwork):
```

Comment:
Do we still need this if statement in clean()?

Reply:
It's replaced by `self.prefix is not None` below - `self.prefix` is always either an IPNetwork or None:

```py
    @property
    def prefix(self):
        if self.cidr_str:
            return netaddr.IPNetwork(self.cidr_str)
        return None
```